.github/workflows/dev-deploy.yaml

```
@@ -34,36 +42,47 @@
                   if grep -q -E '^(fitnessllm_dataplatform/|Dockerfile|pyproject.toml|poetry.lock)' changed_files.txt; then
-                    echo "docker_build_needed=true" >> $GITHUB_OUTPUT
+                    echo "changes_detected=true" >> $GITHUB_OUTPUT
                   else
-                    echo "docker_build_needed=false" >> $GITHUB_OUTPUT
+                    echo "changes_detected=false" >> $GITHUB_OUTPUT
                   fi
 
             - name: Set up Python
+              if: steps.check_changes.outputs.changes_detected == 'true' || (github.event_name == 'workflow_dispatch' && github.event.inputs.force_deploy
+                  == 'true')
               uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)